core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -187,16 +188,20 @@ private void createZkNamespace() {
     String schemaRegistryNamespace = "/" + schemaRegistryZkNamespace;
     schemaRegistryZkUrl = zkConnForNamespaceCreation + schemaRegistryNamespace;
 
-    ZkClient zkClientForNamespaceCreation = new ZkClient(zkConnForNamespaceCreation,
-                                                         zkSessionTimeoutMs, zkSessionTimeoutMs,
-                                                         new ZkStringSerializer());
+    ZkUtils zkUtilsForNamespaceCreation = ZkUtils.apply(
+        zkConnForNamespaceCreation,
+        zkSessionTimeoutMs, zkSessionTimeoutMs,
+        JaasUtils.isZkSecurityEnabled(System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)));
     // create the zookeeper namespace using cluster.name if it doesn't already exist
-    ZkUtils.makeSurePersistentPathExists(zkClientForNamespaceCreation, schemaRegistryNamespace);
+    zkUtilsForNamespaceCreation.makeSurePersistentPathExists(
```

Comment:
I'm a bit confused about why this is being called here. Why is it the responsibility of schema registry to do it? I'm trying to think of how a cluster is supposed to be rolled out.